### PR TITLE
Fix broken signurl error message.

### PR DESCRIPTION
**Pull request by NickGoog**
Fixes http://b/167584551

### CODE REVIEW

**Comment by NickGoog** [comment]
`client_email` is used later in the function, so please check the code branches to make sure I'm not introducing any unwanted behavior in URL generation (I actually hope this will fix URL generation for a separate issue).

File: gslib/gcs_json_credentials.py [hunk 1]
```diff
@@ -133,6 +133,12 @@ def GetCredentialStoreKey(credentials, api_version):
   return '-'.join(key_parts)
 
 
+def GetImpersonateServiceAccount():
```
Comment (dilipped) [inline gslib/gcs_json_credentials.py:136]: This does not cover the service account scenario.

We should be instead using https://github.com/GoogleCloudPlatform/gsutil/blob/3b61bf0e6188f65f78c72c79ea3cb69e9c61da4b/gslib/gcs_json_api.py#L314

But that method is inside gcs_json_api, which means we will have to add an API method that exposes this method.

**Review by dilipped — CHANGES_REQUESTED** [review]

File: gslib/gcs_json_credentials.py (hunk 1 above)
Comment (NickGoog) [reply gslib/gcs_json_credentials.py]: Fixed!

**Review by dilipped — APPROVED** [review]
Make sure to run the S3 tests to ensure that nothing is breaking there.

**Comment by dilipped** [comment]
Looks like Travis is failing

**Comment by NickGoog** [comment]
S3 passed. Travis is disappointed I forgot to yapf haha

**Pull request merged**